src/puppetlabs/puppetdb/config.clj

```diff
@@ -156,7 +156,7 @@
    :subname     (format "file:%s;hsqldb.tx=mvcc;sql.syntax_pgs=true" (io/file vardir "db"))})
 
 (defn validate-db-settings
-  "Throws a {:type ::configuration-error :message m} exception
+  "Throws a {:type ::cli-error :message m} exception
```

Comment:
Right, I don't know either way wrt cli-error; I just had those two considerations on my mind when I decided to go with configuration-error, and had imagined I might come back and convert a lot of the other IllegalArgumentExceptions to configuration-error, and then handle that higher up.